### PR TITLE
Test/update compliance

### PR DESCRIPTION
**Pull request by johanfylling**
Updating compliance test specs to latest OPA main (0.42.0)

### CODE REVIEW

File: src/main/clojure/jarl/eval.clj [hunk 1]
```diff
@@ -143,7 +143,7 @@
   (log/tracef "NotStmt - <%s> == <%s>" a-pos b-pos)
   (let [a (state/get-value state a-pos)
         b (state/get-value state b-pos)
-        result (= a b)]
+        result (types/rego-equal? a b)]
```
Comment (anderseknert) [inline src/main/clojure/jarl/eval.clj:146]: Interesting — was this required by the changes in this PR, or just a nice unrelated change?

**Review by anderseknert — APPROVED** [review]

File: src/main/clojure/jarl/eval.clj [hunk 2]
```diff
@@ -143,7 +143,7 @@
   (log/tracef "NotStmt - <%s> == <%s>" a-pos b-pos)
```
Comment (anderseknert) [inline src/main/clojure/jarl/eval.clj:143]: This log line seems wrong, if not changed now.

File: src/main/clojure/jarl/eval.clj (hunk 1 above)
Comment (johanfylling) [reply src/main/clojure/jarl/eval.clj:146]: There was a test case added with an equality check between `1` and `1.0`.

File: src/main/clojure/jarl/eval.clj (hunk 2 above)
Comment (johanfylling) [reply src/main/clojure/jarl/eval.clj:143]: I could have sworn I've already fixed this. Must have been a similar mistake somewhere else 😄 .

File: src/main/clojure/jarl/eval.clj (hunk 1 above)
Comment (anderseknert) [reply src/main/clojure/jarl/eval.clj:146]: Ah, right, I should have known, given how it was me who added those 😆

File: src/main/clojure/jarl/eval.clj (hunk 2 above)
Comment (anderseknert) [reply src/main/clojure/jarl/eval.clj]: 👍😄
